recipes/flex/all/test_package/CMakeLists.txt

```diff
@@ -1,20 +1,7 @@
-cmake_minimum_required(VERSION 3.1)
-project(test_package)
+cmake_minimum_required(VERSION 3.15)
+project(test_package LANGUAGES CXX)
 
-# Find FLEX before `conanbuildinfo.cmake` because that file will let `find_program`
-# look for  executables in host packages (let's hope conan 2.0 fixes this)
 find_package(FLEX REQUIRED)
 
-include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
-conan_basic_setup(TARGETS)
-
-find_package(FLEX REQUIRED)
-
-flex_target(flex_scanner basic_nr.l "${PROJECT_BINARY_DIR}/basic_nr.cpp")
-
-add_executable(${PROJECT_NAME} basic_nr.cpp)
-target_include_directories(${PROJECT_NAME} PRIVATE ${FLEX_INCLUDE_DIRS})
-target_link_libraries(${PROJECT_NAME} PRIVATE ${FLEX_LIBRARIES})
-
-add_executable(test_yywrap test_yywrap.c)
-target_link_libraries(test_yywrap PRIVATE ${FLEX_LIBRARIES})
+add_executable(${PROJECT_NAME} ${PROJECT_BINARY_DIR}/basic_nr.cpp)
+target_link_libraries(${PROJECT_NAME} PRIVATE FLEX::FLEX)
```

Comment:
it's not the canonical way to link flex: see https://cmake.org/cmake/help/latest/module/FindFLEX.html

Reply:
Actually this test package should check:

- these variables are defined after `find_package(FLEX REQUIRED)`:

```
FLEX_FOUND - True is flex executable is found
FLEX_EXECUTABLE - the path to the flex executable
FLEX_VERSION - the version of flex
FLEX_LIBRARIES - The flex libraries
FLEX_INCLUDE_DIRS - The path to the flex headers
```

- we can use `FLEX_TARGET()`